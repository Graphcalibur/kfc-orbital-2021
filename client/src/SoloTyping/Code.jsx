import React from 'react';
import CodeLine from "./CodeLine";

const Code = (props) => {
    return <div className="mt-3">
        {props.code.map((line, i) => (
        <label htmlFor="textInput" className="form-label code text" key={line}>
            <CodeLine
                is_curr_line={props.curr_line_num === i}
                line={line}
                first_wrong={props.first_wrong}
                curr_input_len={props.curr_input_len}
            />
        </label>
        ))}
  </div>
}

export default Code;