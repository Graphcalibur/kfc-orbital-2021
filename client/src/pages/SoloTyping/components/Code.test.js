import React from "react";
import Code from "./Code";
import { create } from "react-test-renderer";

describe("Testing Code component in page SoloTyping", () => {
  test("Test 1: Python Code, no wrong", () => {
    const tree = create(
      <Code
        code={[
          "for i in range(10):",
          '    print("number", i)',
          '    print("number squared": i ** 2)',
        ]}
        curr_line_num={2}
        first_wrong={6}
        curr_input_len={6}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test("Test 2: Python Code, with wrong", () => {
    const tree = create(
      <Code
        code={[
          "for i in range(10):",
          '    print("number", i)',
          '    print("number squared": i ** 2)',
        ]}
        curr_line_num={2}
        first_wrong={3}
        curr_input_len={6}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test("Test 3: C++ Code, with wrong", () => {
    const tree = create(
      <Code
        code={["for (int i = 0; i < 10; ++i)", "    cout << i << endl;", "}"]}
        curr_line_num={1}
        first_wrong={12}
        curr_input_len={12}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  test("Test 3: C++ Code, with wrong", () => {
    const tree = create(
      <Code
        code={["for (int i = 0; i < 10; ++i)", "    cout << i << endl;", "}"]}
        curr_line_num={1}
        first_wrong={1}
        curr_input_len={12}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
