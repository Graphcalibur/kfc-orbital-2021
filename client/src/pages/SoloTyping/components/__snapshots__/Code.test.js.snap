// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Code component in page SoloTyping Test 1: Python Code, no wrong 1`] = `
<div
  className="mt-3"
>
  <label
    className="form-label code text"
  >
    for i in range(10):
  </label>
  <label
    className="form-label code text"
  >
        print("number", i)
  </label>
  <label
    className="form-label code text"
  >
    <span>
          
      <u>
        <span
          style={
            Object {
              "color": "#00cc44",
            }
          }
        >
          print(
        </span>
        <span
          style={
            Object {
              "color": "#ff1a1a",
            }
          }
        >
          
        </span>
        "number squared": i ** 2)
      </u>
    </span>
  </label>
</div>
`;

exports[`Testing Code component in page SoloTyping Test 2: Python Code, with wrong 1`] = `
<div
  className="mt-3"
>
  <label
    className="form-label code text"
  >
    for i in range(10):
  </label>
  <label
    className="form-label code text"
  >
        print("number", i)
  </label>
  <label
    className="form-label code text"
  >
    <span>
          
      <u>
        <span
          style={
            Object {
              "color": "#00cc44",
            }
          }
        >
          pri
        </span>
        <span
          style={
            Object {
              "color": "#ff1a1a",
            }
          }
        >
          nt(
        </span>
        "number squared": i ** 2)
      </u>
    </span>
  </label>
</div>
`;

exports[`Testing Code component in page SoloTyping Test 3: C++ Code, with wrong 1`] = `
<div
  className="mt-3"
>
  <label
    className="form-label code text"
  >
    for (int i = 0; i &lt; 10; ++i)
  </label>
  <label
    className="form-label code text"
  >
    <span>
          
      <u>
        <span
          style={
            Object {
              "color": "#00cc44",
            }
          }
        >
          cout &lt;&lt; i &lt;&lt;
        </span>
        <span
          style={
            Object {
              "color": "#ff1a1a",
            }
          }
        >
          
        </span>
         endl;
      </u>
    </span>
  </label>
  <label
    className="form-label code text"
  >
    }
  </label>
</div>
`;

exports[`Testing Code component in page SoloTyping Test 3: C++ Code, with wrong 2`] = `
<div
  className="mt-3"
>
  <label
    className="form-label code text"
  >
    for (int i = 0; i &lt; 10; ++i)
  </label>
  <label
    className="form-label code text"
  >
    <span>
          
      <u>
        <span
          style={
            Object {
              "color": "#00cc44",
            }
          }
        >
          c
        </span>
        <span
          style={
            Object {
              "color": "#ff1a1a",
            }
          }
        >
          out &lt;&lt; i &lt;&lt;
        </span>
         endl;
      </u>
    </span>
  </label>
  <label
    className="form-label code text"
  >
    }
  </label>
</div>
`;
